### PR TITLE
feat: add APIs for setting an item's watched status

### PR DESCRIPTION
**Pull request by Mossop**
@andrey-yantsen before I add any tests here I wanted to see what you thought of this API. Putting them on `Server` was the easiest option to implement. They could equally be on `MediaItem` itself though that would require using `async_trait` (apparently async functions on traits are likely to be stabilised this year though!).

Another oddity here is that the call does of course change the item's metadata so the item is outdated afterwards. I guess we accept that in general though, an items metadata is only correct at the time you retrieve it.

### CODE REVIEW

File: crates/plex-api/src/server/mod.rs [hunk 1]
```diff
@@ -174,6 +200,43 @@ impl Server {
...
+
+    /// Sets a media items playback position in milliseconds. The server currently ignores any
+    /// positions equal to or less than 60000ms
+    pub async fn update_timeline<M: MediaItem>(&self, item: &M, position: u64) -> Result {
+        let rating_key = item.rating_key();
+        let query = Query::new()
+            .param("key", format!("/library/metadata/{rating_key}"))
+            .param("ratingKey", rating_key)
+            .param("offline", "1")
+            .param("state", "stopped")
```
Comment (andrey-yantsen) [inline crates/plex-api/src/server/mod.rs:231]: What are those? :) Can we have those values as arguments?

**Comment by andrey-yantsen** [comment]
Looks great, thanks! Let's continue hoping for `async traits` later this year and deprecate those methods then :)

**Comment by andrey-yantsen** [comment]
> Another oddity here is that the call does of course change the item's metadata so the item is outdated afterwards.

As for this bit, what do you think about updating metadata in the watched/unwatched functions?

And I just realised, I'm not sure about the function names. How do you feel about `mark_watched`/`mark_unwatched`? Or even `scrobble`/`unscrobble`, like in the API?

**Comment by Mossop** [comment]
> > Another oddity here is that the call does of course change the item's metadata so the item is outdated afterwards.
> 
> As for this bit, what do you think about updating metadata in the watched/unwatched functions?

We can do. It just means taking the item as &mut and updating the `MediaItem` trait to allow altering the held metadata. I don't have a strong opinion on that really. Another option could be having the functions return the updated item.

> And I just realised, I'm not sure about the function names. How do you feel about `mark_watched`/`mark_unwatched`? Or even `scrobble`/`unscrobble`, like in the API?

I like `mark_(un)watched`. scrobble is more of a technical term that assumes certain knowledge and so it's less clear IMO.

**Comment by andrey-yantsen** [comment]
> Another option could be having the functions return the updated item.

Yeah, I like it more than `&mut`. Would you mind rewriting the code that way?

File: crates/plex-api/src/server/mod.rs (hunk 1 above)
Comment (Mossop) [reply crates/plex-api/src/server/mod.rs:231]: We could pass the state param, it appears to accept "paused", "buffering", "stopped" and "playing". But it's not clear to me what effect those actually have, I didn't see any obvious difference to how the server behaves.

The offline param is what gets sent by mobile clients when syncing playback that happened while offline. It seemed that without this there is extra work to do, you have to create a playback queue to tell the server you are playing an item, then updated the timeline.

Comment (andrey-yantsen) [reply crates/plex-api/src/server/mod.rs:231]: Why does it always have to be so complicated with Plex 🤦‍♂️ 

Ok, let's start with the current implementation. We'll always have a chance to rewrite it later.

**Comment by andrey-yantsen** [comment]
I still need to look at the code, but I noticed you added a new video.

Some time ago, I added two new files — `testsrc_720p_h264_v1.mkv` and `testsrc_720p_h264_v2.mkv`; they're both 5 minutes long. Do you think you could use those?

**Comment by Mossop** [comment]
> Some time ago, I added two new files — `testsrc_720p_h264_v1.mkv` and `testsrc_720p_h264_v2.mkv`; they're both 5 minutes long. Do you think you could use those?

Oh I could have sworn those were only 5 seconds long! Sure.

**Review by andrey-yantsen — APPROVED** [review]

**Pull request merged**